mmcv/runner/hooks/logger/wandb.py

```diff
@@ -83,6 +94,10 @@ def before_run(self, runner) -> None:
             self.wandb.init(**self.init_kwargs)  # type: ignore
         else:
             self.wandb.init()  # type: ignore
+        if self.define_metric_cfg is not None:
+            for metric, summary in self.define_metric_cfg.items():
```

Comment:
Thanks @okotaku for adding this feature to this base wandb logger. :) A couple of comments.
 
- We should ensure that the summary string passed as the value of `define_metric_cfg` dict should be in `["min", "max", "mean" ,"best", "last", "none"]`. If this is not the case, the user should get a warning, but it should not error out.

- Is there a way to ensure that the `metric` is part of the metrics that will be logged to W&B. I think you can access the metrics by doing this: `tags = self.get_loggable_tags(runner)`. Even if the `metric` is spelled wrong or wrongly provided, `wandb.define_metric` will not throw error but a check will be useful for users imo.

Reply:
I think 2nd one is difficult.
The before run 'tags', train 'tags', val 'tags' are different from each other.